frontend/containers/SignIn/SignIn.jsx

```diff
@@ -0,0 +1,187 @@
+
+    if (nextUser.status === Constants.STATUS_NOT_FOUND) {
+      this.setState({error: 'Authentication API unreachable'})
+    } else if (hasFailed) {
+      this.setState({error:`${isPasswordReset ? 'Passwords don\'t match or invalid token' : 'Email not found or password incorrect'}.`})
+    } else {
+      this.setState({error: false})
+    }
+  }
+
+  render() {
+    const {state, actions, setPagetTitle} = this.props
+    const hasConnectionIssues = state.app.networkStatus !== Constants.NETWORK_OK
+    const {formDataIsValid, error, isPasswordReset} = this.state
+
+    setPagetTitle('Sign In')
+
+    return (
+      <div class={styles.wrapper}>
+        <div class={styles.overlay}>
+          <div class={styles.container}>
+            <form
+              action="/profile"
+              method="POST"
+              onSubmit={this.handleSignIn.bind(this)}
+            >
+              <img class={styles.logo} src="/images/publish.png" />
+
+              {error &&
+                <Banner>{error}</Banner>
+              }
+
+              <div class={styles.inputs}>
+                {!isPasswordReset && (
+                  <div class={styles.input}>
+                    <Label label="Email">
+                      <TextInput
+                        placeholder="Your email address"
+                        validation={this.validation.email}
+                        onChange={this.handleInputChange.bind(this, 'email')}
+                        onKeyUp={this.handleInputChange.bind(this, 'email')}
+                        value={this.state.email}
+                      />
+                    </Label>
+                  </div>
+                )}
+
+                <div class={styles.input}>
+                  <Label label={isPasswordReset ? 'New Password' : 'Password'}>
+                    <TextInput
+                      type="password"
+                      placeholder={isPasswordReset ? 'Your new Password' : 'Your Password'}
+                      onChange={this.handleInputChange.bind(this, 'password')}
+                      value={this.state.password}
+                    />
+                  </Label>
+                </div>
+                {isPasswordReset && (
+                  <div class={styles.input}>
+                    <Label label="Confirm new Password">
+                      <TextInput
+                        type="password"
+                        placeholder="Confirm new Password"
+                        onChange={this.handleInputChange.bind(this, 'passwordConfirm')}
+                        value={this.state.passwordConfirm}
+                      />
+                    </Label>
+                  </div>
+                )}
+              </div>
+
+              <Button
+                accent="system"
+                disabled={hasConnectionIssues || (!formDataIsValid && !isPasswordReset)}
+                type="submit"
+                label={isPasswordReset ? 'Reset password': 'Sign In'}
```

Comment:
This goes back to a previous comment I added, but I still don't get why we need a `label` prop in `<Button>`. Why not just:

```jsx
<Button
  accent="system"
  disabled={hasConnectionIssues || (!formDataIsValid && !isPasswordReset)}
  type="submit"
>{isPasswordReset ? 'Reset password': 'Sign In'}</Button>
```